src/shared/Microsoft.Git.CredentialManager/HttpClientFactory.cs

```diff
@@ -20,17 +22,55 @@ public interface IHttpClientFactory
         /// Creating a new <see cref="HttpClient"/> consumes one free socket which may not be released
         /// by the Operating System until sometime after the client is disposed, leading to possible free
         /// socket exhaustion.
+        /// <para/>
+        /// The client instance is configured for use behind a proxy using Git's http.proxy configuration
+        /// for the local repository (the current working directory), if found.
         /// </remarks>
         /// <returns>New client instance with default headers.</returns>
         HttpClient CreateClient();
     }
 
     public class HttpClientFactory : IHttpClientFactory
     {
+        private readonly ITrace _trace;
+        private readonly ISettings _settings;
+
+        public HttpClientFactory(ITrace trace, ISettings settings)
+        {
+            EnsureArgument.NotNull(trace, nameof(trace));
+            EnsureArgument.NotNull(settings, nameof(settings));
+
+            _trace = trace;
+            _settings = settings;
+        }
+
         public HttpClient CreateClient()
         {
-            // Initialize a new HttpClient
-            var client = new HttpClient();
+            HttpClientHandler handler;
+
+            if (TryCreateProxy(out IWebProxy proxy))
+            {
+                handler = new HttpClientHandler();
+            }
+            else
+            {
+                handler = new HttpClientHandler
+                {
+                    Proxy = proxy,
+                    UseProxy = true
+                };
+            }
+
+            if (!_settings.IsCertificateVerificationEnabled)
+            {
+#if NETFRAMEWORK
+                ServicePointManager.ServerCertificateValidationCallback = (req, cert, chain, errors) => true;
```

Comment:
This is interesting, based on the type of .Net we have to do different settings here?  How fun was this one to find :-)?

Reply:
A quick Internet search produced some help on how to configure the proxy settings in both the .NET Core and Framework HTTP stacks.